### PR TITLE
Allow option to load in encodings ref from local file path instead of external call

### PR DESCRIPTION
**Pull request by nanditamohan**
This is to enable private network scenarios in AzureML RAG, where external public network calls (e.g. to _https://openaipublic.blob.core.windows.net_) will be restricted. In this case, we would refer tiktoken to use the encodings file from the image itself.

### CODE REVIEW

File: tiktoken/registry.py [hunk 1]
```diff
@@ -60,7 +60,12 @@ def get_encoding(encoding_name: str) -> Encoding:
             raise ValueError(f"Unknown encoding {encoding_name}")
 
         constructor = ENCODING_CONSTRUCTORS[encoding_name]
-        enc = Encoding(**constructor())
+        if encoding_name == "gpt2":
+            enc = Encoding(**constructor(local_vocab_bpe_path = local_vocab_bpe_path, local_encoding_path=local_encoding_path))
```
Comment (tot0) [inline tiktoken/registry.py:64]:
```suggestion
            enc = Encoding(**constructor(local_vocab_bpe_path=local_vocab_bpe_path, local_encoding_path=local_encoding_path))
```

**Review by tot0 — APPROVED** [review]

**Comment by nanditamohan** [comment]
@hauntsaninja could you please help review these changes? Thank you!

**Comment by nanditamohan** [comment]
@hauntsaninja ping on this PR - please help review as we have dependency on this change for an upcoming deadline.

**Comment by Mansoorkhango** [comment]
HOW ARE YOU

**Review by hauntsaninja — COMMENTED** [review]
Thanks for the PR! Sorry for the delay, I was out on vacation / busy with other things (maintaining this library is not the focus of my role at work).

This change makes the API quite complex.

For this to work, you need to control the call sites of `tiktoken.get_encoding(..., in_order_to_specify_additional_args)`. In that case, why not just directly instantiate the Encoding object? (see example in https://github.com/openai/tiktoken#extending-tiktoken )

I see two potential counter-arguments:
1) the first is that you're relying on `tiktoken.get_encoding`'s cache. That is, you only control one callsite for `tiktoken.get_encoding` and you call it early with your args, and then everything else gets it. This feels pretty magical; I'd prefer to add a more explicit `tiktoken.set_encoding` API in this case.
2) That you'd actually be happy with https://github.com/openai/tiktoken#extending-tiktoken, but private attributes are scary. In which case I'd prefer to make those private attributes public.

Thoughts?

**Pull request closed without merging**